### PR TITLE
Integrate new homepage UI with server

### PR DESCRIPTION
**Pull request by pedroperson**
First pull request, added new images, css, js, and font folders. Added files in templates folder. 

First pull with new design.

Don't know if this is going to work.

Design is not finalized yet,

### CODE REVIEW

File: templates/new_conversation.html [hunk 1]
```diff
@@ -1,29 +1,291 @@
...
+
+    <script type="text/javascript">
+        // GLOBAL VARIABLES
+        var window_height = window.innerHeight;
+        var window_width = window.innerWidth;
+
+        // PRE-LOAD STYLING
+        document.getElementById("hero-section").style.minHeight = window_height+'px';
+
+        window.onscroll = function(){
```
Comment (dzbarsky) [inline templates/new_conversation.html]: isn't this a copy-paste of the other one? this is why we should have them in JS files and reuse the code

File: static/css/main.css [hunk 2]
```diff
@@ -0,0 +1,2041 @@
...
+  display: flex;
+  width: 900px;
+  max-width: 100%;
+  -webkit-box-pack: justify;
+  -webkit-justify-content: space-between;
+  -ms-flex-pack: justify;
+  justify-content: space-between;
+}
+
+.text-block-6 {
```
[the remark this answers is not in the document]
Comment (pedroperson) [reply static/css/main.css]: all the generic class names were generated as i went by Webflow. I'll be going over them today as I make the design changes Peter and Adeel mentioned last meeting

File: static/css/main.css [hunk 3]
```diff
@@ -0,0 +1,2041 @@
...
+  font-weight: 900;
+  font-style: normal;
+}
+@font-face {
+  font-family: 'Sf_reg';
+  src: url('../fonts/SF-Pro-Display-Regular.otf') format('opentype');
+  font-weight: 400;
+  font-style: normal;
+}
+@font-face {
```
[the remark this answers is not in the document]
Comment (pedroperson) [reply static/css/main.css]: The only other way i know of is through javascript but I wouldn't be much cleaner and the extra server request may not be worth it. Maybe we can have a "fonts.css" file that just references fonts and we just never touch it

File: static/js/anime.min.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+/*
+ 2017 Julian Garnier
```
[the remark this answers is not in the document]
Comment (pedroperson) [reply static/js/anime.min.js]: are you talking about the anime.js library? 
I used it here cus I had already written with it for another program so I just did a quick copy pasta to speed up the progress. It is just easier to describe and animate than the standard css. But we don't do anything too complex yet so I'll see if I can achieve the current animations without it.

File: templates/conversation.html [hunk 5]
```diff
@@ -1,20 +1,50 @@
...
+   
+                {{template "conversation" .Conversation}}
+    </div>
+
+
+
+
+    <script type="text/javascript">
+        // GLOBAL VARIABLES
+        var window_height = window.innerHeight;
```
[the remark this answers is not in the document]
Comment (pedroperson) [reply templates/conversation.html]: fixed.
do you want me to do the html ids like this too? (since they can be referenced to directly from the javacript)

File: templates/new_conversation.html (hunk 1 above)
Comment (pedroperson) [reply templates/new_conversation.html]: it is. I moved the current general functions to the conversation.js file.

File: templates/fragments/select.html [hunk 6]
```diff
@@ -1,9 +1,9 @@
...
-
-    {{ range $i, $o := .Options }}
-        <div class="select__item" onclick="javascript:callEndpointWithArgs({{.Endpoint}}, {{$convoId}}, {{$i}})"> {{.Value}} </div>
-    {{ end }}
-</div>
-{{ end }}
+{{ define `select` }}
+<div class="select">
+    {{ $convoId := .ConvoId }}
+
```
[the remark this answers is not in the document]
Comment (pedroperson) [reply templates/fragments/select.html]: Ill look into it. Uploading the new files I noticed that github seems to not like files that dont end with a new line (or at least they put a red icon when it happens) so I put a line at the end of all the files, just in case this is the problem you were talking about)

[the remark this answers is not in the document]
Comment (dzbarsky) [reply templates/fragments/select.html]: yeah i saw that, i'm not sure why they complain but it doesn't hurt to end in newline.

So the real thing i'm trying to figure out is why git is claiming this file has every line changed when you didn't actually change it. I'm not sure if this is due to how you upload the diff or due to how windows handles lines endings. See https://help.github.com/articles/dealing-with-line-endings/

File: render.go [hunk 7]
```diff
@@ -1,206 +1,207 @@
-package main
-
-import (
-	"html/template"
-	"io"
-
```
Comment (dzbarsky) [inline render.go]: please undo this change, style is to separate standard library imports from project-specific ones by newline

File: render.go [hunk 8]
```diff
@@ -1,206 +1,207 @@
...
+
+	if convoModel.Step <= model.AskPregnancyStatus {
+		convo.CurrentQuestion = question{
+			ConvoId: convoModel.Id,
+			Options: pregnancyOptions,
+		}
+		return convo
+	}
+
+	if len(convoModel.Questions2)+len(convoModel.MultiselectQuestions)+len(convoModel.PossibleChiefSymptoms) == 0 {
```
Comment (dzbarsky) [inline render.go]: please rebase your changes against latest master. btw, this is why uploading entire file is not a good idea, you are effectively reverting my other changes to this file. whereas if you used normal git flow, git would know you haven't modified anything around this line and merge your change above with what I did, automatically.

File: static/css/main.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (pedroperson) [reply static/css/main.css]: We defnitely won't be using all the fonts on the folder right now. I just didn't cut it down from what Peter had sent me.
I made the fonts.css file.

File: templates/fragments/select.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (pedroperson) [reply templates/fragments/select.html]: It is probably due to how i uploaded the fil

File: templates/conversation.html [hunk 9]
```diff
@@ -1,20 +1,28 @@
...
-    </div>
-</body>
-</html>
+<html>
+  {{template "header.html"}}
+<body>
+ 
+
+  <!--  TOP BANNER -->
+    <div class="nav" id="nav_bar">
```
[the remark this answers is not in the document]
Comment (pedroperson) [reply templates/conversation.html]: The idea is to not use any hyphens, only use underscore for all html classes and ids. but I can change that today as I go over the classes any way

File: templates/fragments/conversation.html [hunk 10]
```diff
@@ -1,45 +1,47 @@
...
+                </div>
+            {{ end}}
+        </div>
+    {{ end}}
+
+    {{ if .NeedToAskAge }}
+        <div class="form-group ">
+            <!-- TODO: input type=number has weird styling on arrows -->
+            <input class="form-control" value="" type="text">
+            <div style=" margin-right:6px; height:50%;min-width:30px;background-image: url(static/images/x_icon.png); background-position: center center; background-repeat: no-repeat;background-size: auto 100%; cursor:pointer;" onclick="clear_this_textbox(event)"></div>
```
[the remark this answers is not in the document]
Comment (pedroperson) [reply templates/fragments/conversation.html]: yes. the css is not right yet,, but thats where it got it from

File: templates/fragments/header.html [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+	
+    
+    <meta charset="utf-8">
+    <title>Checkup - by Lytic</title>
+    <meta content="width=device-width, initial-scale=1" name="viewport">
+
+    <link href="css/normalize.css" rel="stylesheet" type="text/css">
+
+    <link rel="stylesheet" type="text/css" href="static/css/new_conversation.css">
+	<link rel="stylesheet" type="text/css" href="static/css/conversation.css">
```
[the remark this answers is not in the document]
Comment (pedroperson) [reply templates/fragments/header.html]: my tab is set to 4 space too. I can just indent with spaces from now on

File: render.go (hunk 7 above)
Comment (pedroperson) [reply render.go]: done

File: templates/fragments/conversation.html (hunk 10 above)
[the remark this answers is not in the document]
Comment (dzbarsky) [reply templates/fragments/conversation.html]: ok gotcha

File: templates/fragments/header.html (hunk 11 above)
[the remark this answers is not in the document]
Comment (dzbarsky) [reply templates/fragments/header.html]: oh i got it backwards, actually I indented these with tabs originally and then switched to spaces. my bad!

File: templates/fragments/select.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (dzbarsky) [reply templates/fragments/select.html]: ok let's get that sorted out and hopefully this gets cleaned up :)

**Comment by dzbarsky** [comment]
Closing this one out in favor of the other copy
